include/bmi/Bmi_Py_Adapter.hpp

```diff
@@ -236,24 +236,31 @@ namespace models {
                  * this might cause trouble on certain systems, since (depending on the particular sizes of types) that
                  * could produce duplicate "case" values.
                  */
-                //TODO: include other numpy type strings, https://numpy.org/doc/stable/user/basics.types.html
-                if ( (py_type_name == "int" || py_type_name == "int16") && item_size == sizeof(short)) {
+                if (item_size == sizeof(signed char) && (py_type_name == "int8" || py_type_name == "numpy.int8")) {
```

Comment:
From the documentation, this should cover `numpy.byte`, and `ubyte` in the unsigned case.